src/compute/src/render/continual_task.rs

```diff
@@ -495,18 +497,16 @@ impl<K: Ord, V: Ord, D: Semigroup> SinkState<K, V, Timestamp, D> {
         // advancing timestamps, consolidating, and filtering out anything at
         // future timestamps.
         let as_of = &[write_ts.clone()];
-        for (_, t, _) in &mut self.to_append {
+        for ((_, t), _) in self.to_append.as_mut() {
             t.advance_by(AntichainRef::new(as_of))
         }
         // TODO(ct): Metrics for vec len and cap.
-        consolidate_updates(&mut self.to_append);
-        // TODO(ct): Resize the vec down if cap >> len? Or perhaps `Correction`
-        // might already be very close to what we need.
+        self.to_append.consolidate();
```

Comment:
One thing I realized after I typed this up is that AFAICT ConsolidatingVec will happily reduce the cap down to zero. CTs will frequently reduce len down to 0 and probably want to keep some minimum capacity to avoid thrashing as soon as we put something in it again

Reply:
Good point! Should be easy to introduce a minimum capacity and enforce it in `ConsolidatingVec::consolidate`.

Reply:
Done. Kept it configurable and gave MVs the previous behavior, so I didn't have to reason about whether this would be scary to roll out.